### PR TITLE
(#13463) zlib: Bump to 1.2.13

### LINKED ISSUE #13463: [request] zlib/1.2.13
### Package Details
  * Package Name/Version: **zlib/1.2.13**
  * Changelog: **https://www.zlib.net/ChangeLog.txt**

```
Changes in 1.2.13 (13 Oct 2022)
- Fix configure issue that discarded provided CC definition
- Correct incorrect inputs provided to the CRC functions
- Repair prototypes and exporting of new CRC functions
- Fix inflateBack to detect invalid input with distances too far
- Have infback() deliver all of the available output up to any error
- Fix a bug when getting a gzip header extra field with inflate()
- Fix bug in block type selection when Z_FIXED used
- Tighten deflateBound bounds
- Remove deleted assembler code references
- Various portability and appearance improvements
```

PR follows

### PR DESCRIPTION
**Pull request by miklelappo**
Especially important to fix CVE 2022 37434

Specify library name and version:  **zlib/1.2.13**

```
Changes in 1.2.13 (13 Oct 2022)
- Fix configure issue that discarded provided CC definition
- Correct incorrect inputs provided to the CRC functions
- Repair prototypes and exporting of new CRC functions
- Fix inflateBack to detect invalid input with distances too far
- Have infback() deliver all of the available output up to any error
- Fix a bug when getting a gzip header extra field with inflate()
- Fix bug in block type selection when Z_FIXED used
- Tighten deflateBound bounds
- Remove deleted assembler code references
- Various portability and appearance improvements
```

https://github.com/madler/zlib/issues/686

Closes #13463 
---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by mlappo — DISMISSED** [review]

**Review by antonkesy — DISMISSED** [review]

**Comment by miklelappo** [comment]
CI wakeup

**Comment by miklelappo** [comment]
Wake CI

File: recipes/zlib/all/conandata.yml [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
     url: "https://zlib.net/fossils/zlib-1.2.12.tar.gz"
     sha256: "91844808532e5ce316b3c010929493c0244f3d37593afd6de04f71821d5136d9"
   "1.2.11":
     url: "https://zlib.net/fossils/zlib-1.2.11.tar.gz"
     sha256: "c3e5e9fdd5004dcb542feda5ee4f0ff0744628baf8ed2dd5d66f8ca1197cb1a1"
 patches:
+  "1.2.13":
+    - patch_file: "patches/0002-gzguts-xcode12-compile-fix.patch"
   "1.2.12":
     - patch_file: "patches/0001-fix-cmake.patch"
```
Comment (SpaceIm) [inline recipes/zlib/all/conandata.yml:15]: you likely need this patch as well in 1.2.13

Comment (mlappo) [reply recipes/zlib/all/conandata.yml:15]: This one didn't apply cleanly anymore and the build was fine locally... Let's hope that we don't need it anymore and it's fixed in zlib :-)

Comment (mlappo) [reply recipes/zlib/all/conandata.yml:15]: oh... I think you're indeed right...

Comment (miklelappo) [reply recipes/zlib/all/conandata.yml]: Rebased and fixed

**Review by antonkesy — APPROVED** [review]

**Review by mlappo — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Review by antonkesy — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Comment by prince-chrismc** [comment]
@miklelappo do not spam open and close.

The service has a backlog and you need to be more patient.

Be patient.

**Comment by miklelappo** [comment]
Ok :-) I had an impression with previous patch that this worked out after 24 hours waiting in backlog

**Comment by uilianries** [comment]
@miklelappo For every close/open, a new build is created, but it's added as last job to be executed in the queue. So even if you do it several times, you will need to wait until your turn.

**Comment by miklelappo** [comment]
@uilianries , @prince-chrismc I see. No problem. I think today the builds are way faster than few dazs ago

**Comment by prince-chrismc** [comment]
> I think today the builds are way faster than few dazs ago

There are 50% more workers where it was slow! I am really happy to hear it's noticeable improvement!

**Comment by miklelappo** [comment]
@prince-chrismc definitely noticeable :-) 29 hours for previous PR vs 3 hours (for the last retry) this time :-) And also protobuf PR went through today #13471 in just some hours

**Comment by prince-chrismc** [comment]
For reference about how got that actually is

https://github.com/prince-chrismc/conan-center-index-pending-review/issues/1

graph

**Comment by Neustradamus** [comment]
@miklelappo: Good job!

**Pull request merged**